server.go

```diff
@@ -0,0 +1,169 @@
+package weather
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"time"
+	"weather/api"
+)
+
+// struct to config new server
+type Config struct {
+	Port     int
+	LogLevel string
+}
+
+// struct for query string parameters to pass
+// to OWM api
+type UrlParameters struct {
+	City  string
+	Units string
+}
+
+// struct for json
+type Weather struct {
```

Comment:
Isn't this duplicating code in the `api` package? Why don't we get rid of the `api` package and put everything in `package weather`?

Reply:
Moved api package over to weather package and removed duplicate Weather struct